library/firewall_lib_facts.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016,2017,2020,2021 Red Hat, Inc.
+# Reusing some firewalld code
+# Authors:
+# Brennan Paciorek <bpaciore@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: firewall_lib_facts
+short_description: module for firewall role
+requirements:
+  - python3-firewall or python-firewall
+description:
+  Get firewalld configuration information, including default settings
+  and custom settings. There are no options for this module
+options:
+  detailed:
+    description:
+      return more in depth settings for default subdictionary of
+      firewall_config. Matches the structure of the custom
+      subdictionary if this is active.
+    type: bool
+    required: false
+    default: false
+author: Brennan Paciorek (@BrennanPaciorek)
+"""
+
+RETURN = """
+changed:
+  description:
+    whether anything was changed, most likely false
+  type: bool
+  returned: always
+  sample: False
+ansible_facts:
+  description:
+    any facts that are returned via this module running
+    currently only gets firewalld configuration
+    when calling the module
+  type: dict
+  returned: always
+  sample: {
+      "config": {
+          "default": {},
+          "custom": {},
+      },
+  }
+"""
+
+EXAMPLES = """
+# Run with no parameters to gather facts
+firewall_lib_facts:
+"""
```

Comment:
is EXAMPLES necessary? The user will be accessing this module via the firewall var.